cmd/juju/application/addrelation_test.go

```diff
@@ -1,189 +1,97 @@
-// Copyright 2012, 2013 Canonical Ltd.
+// Copyright 2012 Canonical Ltd.
 // Licensed under the AGPLv3, see LICENCE file for details.
 
 package application
 
 import (
+	"github.com/juju/errors"
+	"github.com/juju/testing"
 	jc "github.com/juju/testing/checkers"
 	gc "gopkg.in/check.v1"
 
-	jujutesting "github.com/juju/juju/juju/testing"
-	"github.com/juju/juju/testcharms"
-	"github.com/juju/juju/testing"
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/params"
+	coretesting "github.com/juju/juju/testing"
 )
 
 type AddRelationSuite struct {
-	jujutesting.RepoSuite
-	testing.CmdBlockHelper
+	testing.IsolationSuite
+	mockAPI *mockAddAPI
 }
 
 func (s *AddRelationSuite) SetUpTest(c *gc.C) {
-	s.RepoSuite.SetUpTest(c)
-	s.CmdBlockHelper = testing.NewCmdBlockHelper(s.APIState)
-	c.Assert(s.CmdBlockHelper, gc.NotNil)
-	s.AddCleanup(func(*gc.C) { s.CmdBlockHelper.Close() })
+	s.IsolationSuite.SetUpTest(c)
+	s.mockAPI = &mockAddAPI{
+		addRelationFunc: func(endpoints ...string) (*params.AddRelationResults, error) {
+			// At the moment, cmd implementation ignores the return values,
+			// so nil is an acceptable return for testing purposes.
+			return nil, nil
```

Comment:
if you add a testing.Stub to the suite, and return .NextErr() here, you can get away without redefining s.mockAPI in the error test below, and you can use the .CheckCall method to check the args passed in the success case